libs/language-server/src/lib/ast/expressions/typeguards.ts

```diff
@@ -55,6 +58,16 @@ export const COLLECTION_TYPEGUARD: InternalValueRepresentationTypeguard<
   return Array.isArray(value);
 };
 
-export function isEveryValueDefined<T>(array: (T | undefined)[]): array is T[] {
-  return array.every((value) => value !== undefined);
-}
+export const INVALID_TYPEGUARD = (value: unknown): value is InvalidError => {
+  return value instanceof Error && value.name === 'InvalidError';
+};
+
+export const MISSING_TYPEGUARD = (value: unknown): value is MissingError => {
+  return value instanceof Error && value.name === 'MissingError';
+};
+
+export const ERROR_TYPEGUARD = (
+  value: unknown,
+): value is InternalErrorRepresentation => {
+  return INVALID_TYPEGUARD(value) || MISSING_TYPEGUARD(value);
+};
```

Comment:
Would use classic typeguard function naming `is<TypeName>`

Reply:
Seems to be used similarly on other occasions as well... So align with what is there. Potentially, refactor in the future.